### PR TITLE
Add DrawIndirect test

### PR DESCRIPTION
**Pull request by shrekshao**
Add DrawIndirect part for indirect_draw.spec.ts.
(DrawIndexedIndirect is disabled for Chrome OT)

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented in `helper_index.md`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)
    
**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/api/operation/rendering/indirect_draw.spec.ts [hunk 1]
```diff
@@ -1,13 +1,184 @@
...
+While the remaining indirectBuffer is populated with random numbers or draw args
+that draw right top triangle, both, or nothing which will fail the color check.
+The test will check render target to see if only the left bottom area is filled,
+meaning the expected draw args is uploaded correctly by the indirectBuffer and indirectOffset.
+
+Params:
+    - indirectOffset= {0, 4, k * sizeof(args struct), k * sizeof(args struct) + 4}
+    `
+  )
+  .cases(
```
Comment (kainino0x) [inline src/webgpu/api/operation/rendering/indirect_draw.spec.ts:30]: nit: I think these should all be subcases.
```suggestion
  .subcases(() =>
```

File: src/webgpu/api/operation/rendering/indirect_draw.spec.ts [hunk 2]
```diff
@@ -1,13 +1,184 @@
...
+          Uint32Array.BYTES_PER_ELEMENT,
+          1 * kDrawIndirectParametersSize,
+          1 * kDrawIndirectParametersSize + Uint32Array.BYTES_PER_ELEMENT,
+          3 * kDrawIndirectParametersSize,
+          3 * kDrawIndirectParametersSize + Uint32Array.BYTES_PER_ELEMENT,
+          99 * kDrawIndirectParametersSize,
+          99 * kDrawIndirectParametersSize + Uint32Array.BYTES_PER_ELEMENT,
+        ] as const)
+      )
+      .expand(p => {
```
Comment (kainino0x) [inline src/webgpu/api/operation/rendering/indirect_draw.spec.ts:44]: This `.expand()` appears to only ever return one result, so it doesn't actually end up "expanding" the case list. Could this whole section (computing `indirectBuffer`) just be moved inside the test function?

**Review by kainino0x — APPROVED** [review]
LGTM

File: src/webgpu/api/operation/rendering/indirect_draw.spec.ts (hunk 1 above)
Comment (shrekshao) [reply src/webgpu/api/operation/rendering/indirect_draw.spec.ts]: Do we have some docs for these helpers `cases`, `subcases`, `poptions` ... (or are they names from some common practice?)
I still find a bit confusing to read their source code 😅 
(I now get subcases are without extra dropdown menu)

**Comment by shrekshao** [comment]
Thanks for the review!

File: src/webgpu/api/operation/rendering/indirect_draw.spec.ts (hunk 1 above)
Comment (kainino0x) [reply src/webgpu/api/operation/rendering/indirect_draw.spec.ts]: Probably not, but great idea. I will add some docs.

**Pull request merged**